### PR TITLE
Expose content-type flag to stream creation command

### PR DESCRIPTION
**Pull request by fbiville**
Depends on https://github.com/projectriff/system/pull/28

### CODE REVIEW

File: pkg/riff/commands/stream_create_test.go [hunk 1]
```diff
@@ -46,6 +46,15 @@ func TestStreamCreateOptions(t *testing.T) {
 			},
 			ShouldValidate: true,
 		},
+		{
+			Name: "valid stream",
```
Comment (scothis) [inline pkg/riff/commands/stream_create_test.go]:
```suggestion
			Name: "with content type",
```

move after the `no provider` test

File: pkg/riff/commands/stream_create.go [hunk 2]
```diff
@@ -30,7 +30,8 @@ import (
 type StreamCreateOptions struct {
 	cli.ResourceOptions
 
-	Provider string
+	Provider    string
+	ContentType string
```
Comment (scothis) [inline pkg/riff/commands/stream_create.go]: do we need to apply any validation?

Comment (fbiville) [reply pkg/riff/commands/stream_create.go]: we can check the presence of `/` and check that the first slash is not also the last. I may be missing something but I don't think we can do much more.

**Comment by fbiville** [comment]
> Looks like go.mod is missing

Yep, I need https://github.com/projectriff/system/pull/28 merged first.

**Comment by fbiville** [comment]
Please do not merge as-is, the changes includes a reference to my riff system fork at the moment.

File: pkg/riff/commands/stream_create.go [hunk 3]
```diff
@@ -42,6 +43,14 @@ func (opts *StreamCreateOptions) Validate(ctx context.Context) *cli.FieldError {
 		errs = errs.Also(cli.ErrMissingField(cli.ProviderFlagName))
 	}
 
+	contentType := opts.ContentType
+	if contentType != "" {
+		index := strings.Index(contentType, "/")
```
[the remark this answers is not in the document]
Comment (fbiville) [reply pkg/riff/commands/stream_create.go]: I would personally wait for a second usage before extracting the validation to a separate function but I don't mind doing it now.

**Comment by ericbottard** [comment]
Merged as d488573d

**Pull request closed without merging**